src/State/Hub.php

```diff
@@ -231,21 +234,35 @@ public function startTransaction(TransactionContext $context): Transaction
     {
         $client = $this->getClient();
 
+        $sampleRate = null;
+
+        if (null !== $client) {
+            $sampler = $client->getOptions()->getTracesSampler();
+            if (null !== $sampler) {
+                if (\is_callable($sampler)) {
+                    $sampleRate = $sampler(SamplingContext::getDefault($context));
+                } elseif ($sampler instanceof TracesSamplerInterface) {
+                    $sampleRate = $sampler->sample(SamplingContext::getDefault($context));
+                }
+            }
+        }
+
         // Roll the dice for sampling transaction, all child spans inherit the sampling decision.
-        if (null === $context->sampled) {
+        // And only if $$sampleRate == null because then the traces_sampler wasn't defined and we need to roll the dice
```

Comment:
Can you please rephrase better this comment? I'm having an hard time understanding it